### PR TITLE
[IA-3394] [IA-3431] Add balanced persistent disk to compute + Galaxy modals

### PR DESCRIPTION
**Pull request by n8landolt**
## Description

Add balanced persistent disk to the compute modal and Galaxy modal select menu.

Note - Cromwell appears to not have this same selection.

## Solution

Update the machines.js regionToPrices dict so pricing is available.
Add balanced disk to the pdTypes object.
Include balanced disk to the select menu for compute and Galaxy Modals.
Nice-to-have included fix - select menu will now automatically pop up or down based on available space. Previously, it was down, so users had to scroll to see all options with the addition of another disk type.

## Testing

Manual testing only.
Selected the balanced persistent disk in all 3 offerings.
Viewed the updated cost on the UI after selecting balanced disk.
Created Jupyter, Galaxy, and Rstudio apps using balanced disk.
Reconnected to the balanced disk after deleting an env.

### CODE REVIEW

**Comment by kyuksel** [comment]
> TODO:
Reconnect to the balanced disk after deleting an env.

@n8landolt Given the TODO, I wanted to check if this PR is finalized for reviewing. If not, please convert it to a draft PR so it doesn't get assigned a code owner reviewer and doesn't get triaged.

File: src/data/machines.js
Comment (kyuksel) [inline src/data/machines.js]: Can we be consistent with naming the props; e.g. by having them all be `monthly*DiskPrice` or `*DiskPrice`?

File: src/data/machines.js
Comment (kyuksel) [inline src/data/machines.js]: I'm surprised that the balanced disk prices are not the same as as ssd prices given [this](https://github.com/broadinstitute/dsp-scripts/pull/122#discussion_r881731087).

File: src/data/machines.js
Comment (n8landolt) [reply src/data/machines.js]: Good catch! I'll get on that, requires a change in another repository.

**Comment by n8landolt** [comment]
I still have to make a fix - when deleting an environment, the message "You will still incur charges for your persistent disk at a rate of **unknown** per month"

File: src/data/machines.js
Comment (kyuksel) [reply src/data/machines.js]: Never mind. Got clarification on the thread linked to 😊

**Comment by n8landolt** [comment]
> I still have to make a fix - when deleting an environment, the message "You will still incur charges for your persistent disk at a rate of **unknown** per month"

Appears this is a bug prior to my changes. Created a bug ticket since it will take a little more digging than initially expected.

**Comment by n8landolt** [comment]
> @n8landolt Given the TODO, I wanted to check if this PR is finalized for reviewing. If not, please convert it to a draft PR so it doesn't get assigned a code owner reviewer and doesn't get triaged.

I had no idea that was a thing!!! Thank you!
I've finished my testing, so I wont make it a draft right now. But in the future, I absolutely will.

File: src/components/GalaxyModal.js
Comment (breilly2) [inline src/components/GalaxyModal.js:330]: Nice improvement! I'm very particular about my browser window sizes, but I was able to test this by opening the inspector, which allows me to change the effective height of the browser window without actually resizing the window.

File: src/data/machines.js
Comment (breilly2) [inline src/data/machines.js:255]: Was this an intentional change? It stands out to me as the only change in this file that isn't just adding a `monthlyBalancedDiskPrice`.

**Review by breilly2 — APPROVED** [review]
👍 Looks good to me. I did not verify that the balanced disk prices for each of the regions is correct, but all appear to be somewhere between standard and SSD disk pricing.

File: src/data/machines.js
Comment (n8landolt) [reply src/data/machines.js:255]: Good catch - I would say it's semi intentional. I don't know the cadence in which the machines.js file is updated via the dsp-scripts.
This value was updated from the output of parse_pricing_api.py in dsp-scripts repo. [My edits for the PR are here.](https://github.com/broadinstitute/dsp-scripts/pull/122) It seems like the API got data for t4HourlyPrice instead of NaN.

**Review by Qi77Qi — APPROVED** [review]

**Comment by Qi77Qi** [comment]
reminder don't merge until leo release happens

**Pull request merged**
